priam/src/main/java/com/netflix/priam/cluster/management/IClusterManagement.java

```diff
@@ -15,12 +15,72 @@
  */
 package com.netflix.priam.cluster.management;
 
+import com.netflix.priam.IConfiguration;
+import com.netflix.priam.merics.IMeasurement;
+import com.netflix.priam.merics.IMetricPublisher;
+import com.netflix.priam.scheduler.Task;
+import com.netflix.priam.scheduler.TaskTimer;
+import com.netflix.priam.utils.JMXConnectionException;
+import com.netflix.priam.utils.JMXConnectorMgr;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
 import java.util.List;
 
 /**
  * Created by vinhn on 10/12/16.
  */
-public interface IClusterManagement<T> {
+public abstract class IClusterManagement<T> extends Task {
+    public enum Task {FLUSH, COMPACTION};
+    private static final Logger logger = LoggerFactory.getLogger(IClusterManagement.class);
+    private IMetricPublisher metricPublisher;
+    private Task taskType;
+    private IMeasurement measurement;
+
+    protected IClusterManagement(IConfiguration config, Task taskType, IMetricPublisher metricPublisher, IMeasurement measurement) {
+        super(config);
+        this.taskType = taskType;
+        this.metricPublisher = metricPublisher;
+        this.measurement = measurement;
+    }
+
+    @Override
+    public void execute() throws JMXConnectionException, Exception {
+        JMXConnectorMgr connMgr = null;
+        List<T> result = null;
+
+        try {
+            connMgr = new JMXConnectorMgr(config);
+        }catch (IOException | InterruptedException e){
+            measurement.incrementFailureCnt(1);
+            this.metricPublisher.publish(measurement); //signal that there was a failure
+            throw new JMXConnectionException("Exception during creating JMX connection for operation: " + taskType.name(), e);
+        }
+
+        try {
+            result = manageTask(connMgr);
+            measurement.incrementSuccessCnt(1);
+            this.metricPublisher.publish(measurement); //signal that there was a success
+            logger.info("Successfully finished executing the cluster management task: {} with result: {}", taskType, result);
+        } catch (Exception e) {
+            measurement.incrementFailureCnt(1);
+            this.metricPublisher.publish(measurement); //signal that there was a failure
+            throw new RuntimeException(String.format("Exception during %s.",taskType), e);
+
+        } finally {
+            if (connMgr != null)
```

Comment:
To simplify the resource management, `try-with-resources` would be a better alternative. 
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html